packages/babel-types/src/clone/cloneNode.js

```diff
@@ -66,6 +66,9 @@ export default function cloneNode<T: Object>(node: T, deep: boolean = true): T {
       ...node.extra,
     };
   }
+  if (has(node, "typeAnnotation")) {
```

Comment:
This should be up after line `39`, since I believe it would be the `Identifier` optimization there that causes the problem. The field logic on `43` should already handle it for other types.

This will also need to recursively clone when `deep` is set, like the logic on `45` does.